app/models/file_search.rb

```diff
@@ -72,46 +94,59 @@ def year_range
     all_years
   end
 
-  def results
-    SearchResults.new(files)
-  end
-
   def files
-    params.empty? ? highlighted_files : filtered_files
+    query = {}
+    query[:q] = search_term if search_term
+    query[:f] = filters unless filters.empty?
+    query[:page] = page if page
```

Comment:
Can we call the `search_term`, `filters`, and `page` methods only once without making this method an unreadable mess?